src/arch/util/machine-broadcast.C

```diff
@@ -392,7 +392,10 @@ void CmiFreeNodeBroadcastAllFn(int size, char *msg) {
 
 void CmiWithinNodeBroadcastFn(int size, char* msg) {
   CMI_DEST_RANK(msg) = CmiMyRank();
-  SendToPeers(size, msg);
+#if CMK_ONESIDED_IMPL
```

Comment:
This may be the cause of the hang. When not built with ONESIDED, messages never get forwarded to peers. Did you mean to just enclose the `if` inside of the #if guard?

Reply:
Yes, I just found out that non-ONESIDED layers weren't forwarding the peer messages. I have a patch that I'll push soon. Testing it on both netlrts-smp and mpi-smp right now.